app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, " \
+               f"Health: {self.health}, " \
+               f"Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = False if self.hidden else True
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore_obj: Herbivore) -> None:
```

Comment:
I'm strongly against names like this herbivore_obj. You can just use herbivore

Reply:
Ok) Can I ask you, why? What problems can naming like herbivore_obj cause?